File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java


import java.util.ArrayList;

import static java.util.Collections.max;

class Solution {
    public int solution(int[][] sizes) {
        ArrayList<Integer> big = new ArrayList<Integer>();
        ArrayList<Integer> small = new ArrayList<Integer>();
        int answer = 1;
        for (int i =0; i< sizes.length; i++)
        {
            if (sizes[i][0] > sizes[i][1])
            {
                big.add(sizes[i][0]);
                small.add(sizes[i][1]);
            }
            else
            {
                big.add(sizes[i][1]);
                small.add(sizes[i][0]);
            }
        }
        answer = max(big) * max(small);
        return answer;
    }
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    result = 0
    big = []
    small = []


    for i in range(len(sizes)):
        if sizes[i][0] > sizes[i][1]:
            big.append(sizes[i][0])
            small.append(sizes[i][1])
        else:
            big.append(sizes[i][1])
            small.append(sizes[i][0])

    result = max(big) * max(small)

    return result